### PR TITLE
CHEAT SHEET: FO terminology

### PR DESCRIPTION
**Pull request by jacobmarks**
First draft of cheat sheet introducing the basic terminology in FO. Would love to get this into the 0.19 release! This is higher priority than the filtering and view stages cheat sheets :) 

Once we get this, and the filtering and view stages cheat sheets approved, we can create a new section in the docs for cheat sheets.

-   [ ] No. You can skip the rest of this section.
-   [ ] Yes. Give a description of this change to be included in the release
        notes for FiftyOne users.

(Details in 1-2 sentences. You can just refer to another PR with a description
if this PR is part of a larger change.)

### What areas of FiftyOne does this PR affect?

-   [ ] App: FiftyOne application changes
-   [ ] Build: Build and test infrastructure changes
-   [ ] Core: Core `fiftyone` Python library changes
-   [x] Documentation: FiftyOne documentation changes
-   [ ] Other

### CODE REVIEW

File: docs/source/recipes/index.rst [hunk 1]
```diff
@@ -37,6 +37,13 @@ workflows.
 
 .. Add recipe cards below
 
+.. customcarditem::
+    :header: FiftyOne terminology
+    :description: A brief overview of FiftyOne's key terminology.
+    :link: fiftyone_terminology.html
+    :image: ../_static/images/recipes/pandas_vs_fiftyone.png
```
Comment (brimoor) [inline docs/source/recipes/index.rst:44]: Please add a new thumbnail image for this before merging! 

Tip: make sure you use a square image, or else the image will be squashed/stretched when it is displayed.

File: docs/source/recipes/fiftyone_terminology.rst [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+A cheat sheet introducing you to the key terminology in the world of FiftyOne!
+
+Core components
+_______________
+
+.. list-table::
+   :widths: 20 80
+
+   * - **FiftyOne**
+     - The :ref:`open-source framework <fiftyone-library>`, the core library,
```
Comment (brimoor) [inline docs/source/recipes/fiftyone_terminology.rst:17]: I found that the table cells will automatically line wrap; no need to prepend `|` and manually control the line breaks!

**Review by brimoor — APPROVED** [review]
LGTM! Please update the thumbnail image before merging 🤗

File: docs/source/recipes/fiftyone_terminology.rst (hunk 2 above)
Comment (jacobmarks) [reply docs/source/recipes/fiftyone_terminology.rst:17]: Oh nice!

File: docs/source/recipes/index.rst (hunk 1 above)
Comment (jacobmarks) [reply docs/source/recipes/index.rst:44]: yep! Just waiting on Michelle. She said she will get it in this weekend

**Pull request merged**